### DISCUSSION TITLE
Windows install for use with Selenium

### DISCUSSION THREAD

**Discussion by Auto81** [giggio/node-chromedriver#415] [General]
Having some odd behaviour with using this on Windows machines. It works perfectly on Mac.

Running 'npm i -g chromedriver' succeeds, and then 'chromedriver' in cmd works as expected.

However when Selenium-JVM tried to start up this chromedriver it throws  
```
Cannot run program "C:\Users\Auto81\AppData\Roaming\npm\chromedriver" (in directory "."): 
CreateProcess error=193, %1 is not a valid Win32 application)
```

Upon a closer look the first path location that gets returned seems to be a shell script?
```
C:\Users\Auto81>where chromedriver
C:\Users\Auto81\AppData\Roaming\npm\chromedriver
C:\Users\Auto81\AppData\Roaming\npm\chromedriver.cmd
```

If I delete the script via `del %APPDATA%\npm\chromedriver`, then Selenium is able to start the driver properly.

Only other related item I could find was https://github.com/giggio/node-chromedriver/issues/90

**Comment by giggio** [comment]
When npm installs a global package, it creates that `.cmd` by itself, and it will point to the actual location of the package. That is not directly controlled by the library. The file without extension is intended to be used by Linux-like systems, like Cygwin. The one with `.cmd` is to be used by "normal" Windows. The chromedriver executable that is installed in the directory `node_modules\chromedriver\lib\chromedriver\chromedriver.exe`.

**Reply by Auto81** [reply]
yes, can see the 'node_modules\chromedriver\lib\chromedriver\chromedriver.exe' but would have to add that directory to the environment variables.

Currently having our windows users run `npm i -g chromedriver && del %APPDATA%\npm\chromedriver` to resolve the issue

Is there a way for this library, when process.platform === "win32" maybe, to not create the chromedriver file without the extension? or is that file also not controlled by the library?

**Reply by giggio** [reply]
The thing is, this package is intended to be used by Node.js apps, not Java. When I code in .NET, for example, I use NuGet to get Chromedriver.
I'd recomend that you search for Maven packages that maybe solve the problem the same way. Last time I coded in Java I used webdrivermanager. Check it out: https://github.com/bonigarcia/webdrivermanager

**Reply by Auto81** [reply]
Yeah there is also a built in driver manager in selenium now as of v4.6 too, but both of these have their own issues. Will just stick with the extra manual delete command during install for now
